tests/models/wav2vec2_conformer/test_modeling_wav2vec2_conformer.py

```diff
@@ -872,6 +919,57 @@ def _load_datasamples(self, num_samples):
 
         return [x["array"] for x in speech_samples]
 
+    def check_inference_pretrained(self, model_id, model, input_name, inputs, features_shape):
+        torch.manual_seed(0)
+        mask_time_indices = _compute_mask_indices(
+            features_shape,
+            model.config.mask_time_prob,
+            model.config.mask_time_length,
+            min_masks=2,
+        )
+        mask_time_indices = torch.from_numpy(mask_time_indices).to(torch_device)
+
+        with torch.no_grad():
+            outputs = model(
+                inputs[input_name],
+                attention_mask=inputs["attention_mask"],
+                output_attentions=True,
+                mask_time_indices=mask_time_indices,
+            )
+
+        # compute cosine similarity
+        cosine_sim = torch.cosine_similarity(outputs.projected_states, outputs.projected_quantized_states, dim=-1)
+
+        # retrieve cosine sim of masked features
+        cosine_sim_masked = cosine_sim[mask_time_indices]
+
+        # ... now compare to randomly initialized model
+
+        config = Wav2Vec2ConformerConfig.from_pretrained(model_id)
+        model_rand = Wav2Vec2ConformerForPreTraining(config).to(torch_device).eval()
+
+        with torch.no_grad():
+            outputs_rand = model_rand(
+                inputs[input_name],
```

Comment:
What's the point of passing `input_name` here - is it different for the different models?

Reply:
It is ! `input_values` for the Wav2VecFeatureExtractor and `input_features` for the `SeamlessM4TFeatureExtractor` !